packages/components/src/AtlantisThemeContext/AtlantisThemeContext.tsx

```diff
@@ -0,0 +1,103 @@
+import { darkTokens, tokens } from "@jobber/design";
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import merge from "lodash/merge";
+import {
+  AtlantisThemeContextProviderProps,
+  AtlantisThemeContextValue,
+  THEME_CHANGE_EVENT,
+  Theme,
+  ThemeChangeDetails,
+} from "./types";
+import { updateTheme } from "./updateTheme";
+import { useThemeContextEventQueue } from "./useThemeContextEventQueue";
+import styles from "./AtlantisThemeContext.css";
+
+export const atlantisThemeContextDefaultValues: AtlantisThemeContextValue = {
+  theme: "light",
+  tokens: tokens,
+};
+
+const AtlantisThemeContext = createContext(atlantisThemeContextDefaultValues);
+
+export function AtlantisThemeContextProvider({
+  children,
+  defaultTheme = "light",
+  forceThemeForProvider,
+}: AtlantisThemeContextProviderProps) {
+  const [internalTheme, setInternalTheme] = useState<Theme>(defaultTheme);
+  const providerWrapperRef = useRef<HTMLDivElement>(null);
+  const { isEmpty, dequeueThemeChange, enqueueThemeChange, themeChangeQueue } =
+    useThemeContextEventQueue();
+
+  const handleThemeChange = useCallback(
+    (event: Event) => {
+      const newTheme = (event as CustomEvent<ThemeChangeDetails>).detail.theme;
+      enqueueThemeChange(newTheme);
+    },
+    [enqueueThemeChange],
+  );
+  const updateProviderRef = useCallback((theme: Theme) => {
+    if (!providerWrapperRef.current) return;
+    providerWrapperRef.current.dataset.theme = theme;
+  }, []);
+
+  const currentTokens = useMemo(
+    () => (internalTheme === "dark" ? merge(tokens, darkTokens) : tokens),
+    [internalTheme],
+  );
+
+  useEffect(() => {
```

Comment:
See if these useEffects can be simplified / made more clear as to what they do

Reply:
I don't like that there are some useEffect chains. I am updating it right now to see if I can simplify the control flow